plugins/org.eclipse.osee.framework.core/src/org/eclipse/osee/framework/core/data/OrcsTokenServiceImpl.java

```diff
@@ -437,6 +437,40 @@ public Set<ArtifactTypeToken> getValidArtifactTypes(AttributeTypeToken attribute
       return artifactTypes;
    }
 
+   @Override
+   public List<RelationTypeToken> getValidRelationTypes() {
+      Collection<RelationTypeToken> relationTypes = getRelationTypes();
+      List<RelationTypeToken> invalidRels = new ArrayList<>();
+      //populate convertedRelations if empty
+      if (convertedRelations.isEmpty()) {
+         setConvertedRelations();
+      }
+      //add all rels in shadowRels and remove the ones that have been converted
+      for (Field field : Arrays.asList(ShadowCoreRelationTypes.class.getDeclaredFields())) {
+         try {
+            RelationTypeToken rel = (RelationTypeToken) field.get(null);
```

Comment:
What is this doing?

Reply:
This is for when we start actually converting relations.  Once DefaultHierarchy is converted for example, we would remove the "old" one from what can be returned as a valid relation.